src/tool/cppwinrt/strings/base_collections.h

```diff
@@ -41,8 +41,50 @@ namespace winrt::impl
 
     private:
 
-        T const* m_collection = nullptr;
-        uint32_t m_index = 0;
+        T const* m_collection{};
+        uint32_t m_index{};
+    };
+
+    template <typename T>
+    struct rfast_iterator
+    {
+        using iterator_category = std::input_iterator_tag;
+        using value_type = T;
+        using difference_type = ptrdiff_t;
+        using pointer = T *;
+        using reference = T &;
+
+        rfast_iterator(T const& collection, int64_t const index) noexcept :
+            m_collection(&collection),
+            m_index(index)
+        {}
+
+        rfast_iterator& operator++() noexcept
```

Comment:
any reason not to make this bidirectional, or even random access?

Reply:
No reason, I've just not come across a motivating use case. I'll add an issue to address this for both begin/rbegin.